internal/infrastructure/kubernetes/ratelimit/resource.go

```diff
@@ -6,18 +6,40 @@
 package ratelimit
 
 import (
+	"context"
 	"fmt"
 
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	egcfgv1a1 "github.com/envoyproxy/gateway/api/config/v1alpha1"
+	"github.com/envoyproxy/gateway/internal/provider/kubernetes"
 )
 
 const (
 	// RedisSocketTypeEnvVar is the redis socket type.
 	RedisSocketTypeEnvVar = "REDIS_SOCKET_TYPE"
 	// RedisURLEnvVar is the redis url.
 	RedisURLEnvVar = "REDIS_URL"
+	// RedisAuthEnvVar is the redis auth.
+	RedisAuthEnvVar = "REDIS_AUTH"
+	// RedisTLS is the redis tls.
+	RedisTLS = "REDIS_TLS"
+	// RedisTLSClientCertEnvVar is the redis tls client cert.
+	RedisTLSClientCertEnvVar = "REDIS_TLS_CLIENT_CERT"
+	// RedisTLSClientCertFilename is the reds tls client cert file.
+	RedisTLSClientCertFilename = "/certs/tls.crt"
+	// RedisTLSClientKeyEnvVar is the redis tls client key.
+	RedisTLSClientKeyEnvVar = "REDIS_TLS_CLIENT_KEY"
+	// RedisTLSClientKeyFilename is the redis client key file.
+	RedisTLSClientKeyFilename = "/certs/tls.key"
+	// RedisTLSCaCertEnvVar is the redis tls ca cert.
+	RedisTLSCaCertEnvVar = "REDIS_TLS_CACERT"
+	// RedisTLSCaCertFilename is the redis tls ca cert file.
+	RedisTLSCaCertFilename = "/certs/ca.crt"
+	// RedisTLSSkipHostNameVerificationEnvVar is the redis skip hostname verification.
+	RedisTLSSkipHostNameVerificationEnvVar = "REDIS_TLS_SKIP_HOSTNAME_VERIFICATION"
```

Comment:
we shouldnt allow this by default, lets rm this, use can explicitly set this with `env` if they want to